app/src/test/java/com/goldenraven/padawanwallet/data/RepositoryTest.kt

```diff
@@ -0,0 +1,130 @@
+        `when`(sharedPrefsMock.getString("mnemonic", "No seed phrase saved")).thenReturn("seedPhraseTest")
+        assertEquals("Mnemonic get is incorrect", "seedPhraseTest", repo.getMnemonic())
+    }
+
+    @Test
+    fun `Check if getting Mnemonic is correct when it does not exist`(){
+        `when`(sharedPrefsMock.getString("mnemonic", "No seed phrase saved")).thenReturn(null)
+        assertEquals("Mnemonic get is incorrect", "Seed phrase not there", repo.getMnemonic())
+    }
+
+    @Test
+    fun `Check if faucet was called before`(){
+        `when`(sharedPrefsMock.getBoolean("oneTimeFaucetCallDone", false)).thenReturn(true)
+        assertEquals("Faucet call is wrong", true, repo.wasOneTimeFaucetCallDone())
+    }
+
+    @Test
+    fun `Check if faucet done call is correct`(){
+        repo.offerFaucetCallDone()
+        verify(editorMock, times(1)).putBoolean("offerFaucetCallDone", true)
+    }
+
+    @Test
+    fun `Check if load first time tutorials is correct`(){
+        `when`(sharedPrefsMock.getBoolean("firstTimeTutorialsDone", true)).thenReturn(true)
+        val loadTutorialDoneTest = repo.loadTutorialsDone()
+        verify(editorMock, times(9)).putBoolean(anyString(), anyBoolean())
+        assertEquals(mutableMapOf(
+            "e1" to false,
+            "e2" to false,
+            "e3" to false,
+            "e4" to false,
+            "e5" to false,
+            "e6" to false,
+            "e7" to false,
+            "e8" to false,
+        ), loadTutorialDoneTest)
+    }
+
+    @Test
+    fun `Check if load future tutorials is correct`(){
+        `when`(sharedPrefsMock.getBoolean("firstTimeTutorialsDone", true)).thenReturn(false)
+        `when`(sharedPrefsMock.getBoolean("e1", false)).thenReturn(true)
+        `when`(sharedPrefsMock.getBoolean("e2", false)).thenReturn(true)
+        `when`(sharedPrefsMock.getBoolean("e3", false)).thenReturn(true)
+        `when`(sharedPrefsMock.getBoolean("e4", false)).thenReturn(true)
+        `when`(sharedPrefsMock.getBoolean("e5", false)).thenReturn(true)
+        `when`(sharedPrefsMock.getBoolean("e6", false)).thenReturn(true)
+        `when`(sharedPrefsMock.getBoolean("e7", false)).thenReturn(true)
+        `when`(sharedPrefsMock.getBoolean("e8", false)).thenReturn(true)
+        val loadTutorialDoneTest = repo.loadTutorialsDone()
+        assertEquals(mutableMapOf(
+            "e1" to true,
+            "e2" to true,
+            "e3" to true,
+            "e4" to true,
+            "e5" to true,
+            "e6" to true,
+            "e7" to true,
+            "e8" to true,
+        ), loadTutorialDoneTest)
+    }
+
+    @Test
+    fun `Check if updating tutorials is correct`(){
+        repo.updateTutorialsDone(tutorialNumber = 1)
+        verify(editorMock, times(1)).putBoolean("e1", true)
+    }
+
+    @Test
+    fun `Check if resetting tutorials is correct`(){
+        repo.resetTutorials()
+        verify(editorMock, times(8)).putBoolean(anyString(), anyBoolean())
```

Comment:
Can you explain a bit more what this is doing? Does it just try to apply the `resetTutorial` method 8 times and sees if it throws and error?

Reply:
I am letting Mockito verify if the editor had `.putBoolean(anyString(), anyBoolean())` called 8 times, for example `editor.putBoolean("e1", false)` in this case. I'm not sure if verifying that each variable(e1-e8) should be checked specifically or just in general like what I am doing right now is better though